app/controllers/automated_tests_controller.rb

```diff
@@ -35,14 +35,23 @@ def student_interface
     @student = current_role
     @grouping = @student.accepted_grouping_for(@assignment.id)
 
-    unless @grouping.nil?
+    if @grouping.present?
       @grouping.refresh_test_tokens
       @authorized = flash_allowance(:notice,
                                     allowance_to(:run_tests?,
                                                  current_role,
                                                  context: { assignment: @assignment, grouping: @grouping })).value
+
+      if @assignment.enable_student_tests && !@assignment.non_regenerating_tokens
```

Comment:
oh one more thing, should add a condition for `!@assignment.unlimited_tokens` here as well (matching what's in the view file). Otherwise I suppose it'd be possible for `@assignment.token_period` to be `nil` (or `0`), which wouldn't be good for the logic inside this block